core/error.rs

```diff
@@ -210,6 +211,93 @@ pub struct JsStackFrame {
   pub promise_index: Option<i64>,
 }
 
+/// Applies source map to the given V8 location
+/// (i.e. column number is 0-based).
+fn apply_source_map<'a>(
+  source_mapper: &mut crate::source_map::SourceMapper,
+  file_name: Cow<'a, str>,
+  line_number: i64,
+  column_number: i64,
+) -> (Cow<'a, str>, i64, i64) {
+  match source_mapper.apply_source_map(
+    &file_name,
+    line_number as u32,
+    column_number as u32,
+  ) {
+    SourceMapApplication::Unchanged => {
+      (file_name, line_number, column_number.into())
+    }
+    SourceMapApplication::LineAndColumn {
+      line_number,
+      column_number,
+    } => (file_name, line_number.into(), column_number.into()),
+    SourceMapApplication::LineAndColumnAndFileName {
+      file_name,
+      line_number,
+      column_number,
+    } => (file_name.into(), line_number.into(), column_number.into()),
+  }
+}
+
+/// Parses an eval origin string from V8, returning
+/// the contents before the location,
+/// (the file name, line number, and column number), and
+/// the contents after the location.
+///
+/// # Example
+/// ```ignore
+/// assert_eq!(
+///   parse_eval_origin("eval at foo (bar at (file://a.ts:1:2))"),
+///   Some(("eval at foo (bar at (", ("file://a.ts", 1, 2), "))")),
+/// );
+/// ```
+///
+fn parse_eval_origin(
```

Comment:
Fixes an issue where we weren't actually source mapping the origin for errors from eval. So previously if you had

```ts
// foo.ts



eval("throw new Error()"); // this is line 5
```

You would get

```
Error: Error
    at eval (eval at <anonymous> (file:///foo.ts:2:1), <anonymous>:1:7)
    at file:///foo.ts:5:1
```

Note that the eval part is incorrect `(eval at <anonymous> (file:///foo.ts:2:1)`, it should be line 5 instead of line 2. With this change, you get the right answer.

---

It's pretty minor so I'm fine removing this if it's not worth the complexity

Reply:
I think it's fine, can we have a small unit test for it?